File: LeetCode/1306. Jump Game III.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    bool canReach(vector<int>& arr, int start) {
        if(start < 0 || start >= size(arr) || arr[start] < 0) 
            return false ;
        arr[start] *= -1;
        return !arr[start] || canReach(arr, start + arr[start]) || canReach(arr, start - arr[start]) ;
    }
};

File: LeetCode/1306. Jump Game III.py
###  KNOWLEDGE IS POWER  ###

class Solution(object):
    def canReach(self, arr, start):
        """
        :type arr: List[int]
        :type start: int
        :rtype: bool
        """
        if start < 0 or start >= len(arr) or arr[start] < 0: 
            return False 
        arr[start] *= -1;
        return arr[start] == 0 or self.canReach(arr, start + arr[start]) or self.canReach(arr, start - arr[start])
